### PR TITLE
HTTP message service handles signed entities

### PR DESCRIPTION
**Pull request by ghubertpalo**
## Content
This PR includes a patch that use the HTTP message service to deliver directly HTTP messages for signed entities to the web server. By doing so, it avoids expensive key decode / encode.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [ ] Key commits have useful messages
- PR
  - [X] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Comments
None

## Issue(s)
Relates to #1327

### CODE REVIEW

File: mithril-aggregator/src/database/provider/signed_entity.rs [hunk 1]
```diff
@@ -63,6 +68,63 @@ where
     }
 }
 
+impl TryFrom<SignedEntityRecord> for SnapshotMessage {
+    type Error = StdError;
+
+    fn try_from(value: SignedEntityRecord) -> Result<Self, Self::Error> {
+        #[derive(Deserialize)]
+        struct TmpSnapshot {
```
Comment (dlachaumepalo) [inline mithril-aggregator/src/database/provider/signed_entity.rs:76]: Why not call it `SnapshotArtifact`. Unless it's important to point out that it's temporary.

File: mithril-aggregator/src/database/provider/signed_entity.rs [hunk 2]
```diff
@@ -63,6 +68,63 @@ where
...
+        Ok(snapshot_message)
+    }
+}
+
+impl TryFrom<SignedEntityRecord> for MithrilStakeDistributionMessage {
+    type Error = StdError;
+
+    fn try_from(value: SignedEntityRecord) -> Result<Self, Self::Error> {
+        #[derive(Deserialize)]
+        struct TmpMithrilStakeDistribution {
```
Comment (dlachaumepalo) [inline mithril-aggregator/src/database/provider/signed_entity.rs:106]: Same remark as for the snapshot.

File: mithril-aggregator/src/services/http_message.rs [hunk 3]
```diff
@@ -30,18 +37,35 @@ pub trait HttpMessageService: Sync + Send {
...
     /// Return the message representation of the last N certificates
     async fn get_last_certificates(&self, limit: usize) -> StdResult<CertificateListMessage>;
+
+    /// Return the information regarding the given snapshot
+    async fn get_snapshot_message(
+        &self,
+        signed_entity_id: &str,
+    ) -> StdResult<Option<SnapshotMessage>>;
+
+    /// Return the information regarding the given snapshot
```
Comment (dlachaumepalo) [inline mithril-aggregator/src/services/http_message.rs:47]:
```suggestion
    /// Return the information regarding the given mithril stake distribution
```

File: mithril-aggregator/src/database/provider/signed_entity.rs (hunk 1 above)
Comment (ghubertpalo) [reply mithril-aggregator/src/database/provider/signed_entity.rs:79]: I took the same name as the artifact it replaces `Snapshot` and added the temporary prefix.

**Review by jpraynaud — COMMENTED** [review]
Do we still need to keep the previous message adapters that are not used anymore?

**Review by dlachaumepalo — COMMENTED** [review]
I have no comments other than those raised by @jpraynaud. Otherwise, LGTM

**Comment by ghubertpalo** [comment]
> Do we still need to keep the previous message adapters that are not used anymore?

They are still used mainly in tests in `MessageService` but not only.

File: mithril-aggregator/src/services/http_message.rs [hunk 4]
```diff
@@ -147,4 +247,215 @@ mod tests {
...
+        let snapshot = service.get_snapshot_message("whatever").await.unwrap();
+
+        assert!(snapshot.is_none());
+    }
+
+    #[tokio::test]
+    async fn get_snapshot() {
+        let beacon = fake_data::beacon();
+        let entity = SignedEntity {
+            signed_entity_id: "msd1".to_string(),
```
[the remark this answers is not in the document]
Comment (ghubertpalo) [reply mithril-aggregator/src/services/http_message.rs:265]: Sorry, I don't see the issue here.

[the remark this answers is not in the document]
Comment (dlachaumepalo) [reply mithril-aggregator/src/services/http_message.rs:265]: And `beacon` should be `epoch`

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Pull request merged**